chroot-bin/volume

```diff
@@ -0,0 +1,158 @@
+#!/bin/sh
+# Copyright (c) 2015 The crouton Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This script uses amixer to present an interface similar to the
+# brightness script for crouton.
+
+set -e
+set -u
+
+DEFAULT_VOLUME_DELTA=5
+
+APPLICATION="${0##*/}"
+USAGE="$APPLICATION [set] [0-100]
+$APPLICATION up|down [0-100]
+$APPLICATION mute [set|unset|toggle]
+
+This script changes the volume of the current output
+device, as would changing it with the shortcut keys or
+with the GUI in Chrome OS.
+
+Shortcut invocations:
+    $APPLICATION up     increases by $DEFAULT_VOLUME_DELTA
+    $APPLICATION down   decreases by $DEFAULT_VOLUME_DELTA
+    $APPLICATION mute   sets muted
+    $APPLICATION 0-100  sets volume to given value
+"
+# The control which controls the current output.
+ALSA_CONTROL="Master"
+
+# Tiny sugar coating around amixer for common parameters.
+_amixer() {
+    amixer -Dcras "$@"
+}
+
+set_volume() {
+    local volume="$1"
+    _amixer -q sset $ALSA_CONTROL "$volume"
+}
+
+get_is_muted() {
+    # Get the alsa state of the control.
+    local state="$(_amixer cget "name=$ALSA_CONTROL Playback Switch" | sed -n \
+        -e "/[[:space:]]\+:/s/.*=//p")"
+    #       ^              ^      ^ Print the line.
+    #       ^              ^ Replace everything before the equal sign.
+    #       ^ Search for the line with spaces and a colon (the value).
+
+    # Muted is off (control is off)
+    if [ "$state" = "on" ]; then
+        # Is not muted, so is_muted is false
+        return 1
+    else
+        # Is muted, so is_muted is true
+        return 0
+    fi
+}
+
+toggle_mute() {
+    _amixer -q sset $ALSA_CONTROL toggle
+}
+
+mute() {
+    _amixer -q sset $ALSA_CONTROL mute
+}
+unmute() {
+    _amixer -q sset $ALSA_CONTROL unmute
+}
+
+relative_volume() {
+    local delta="$1"
+    if get_is_muted; then
+        if [ "$delta" -lt 0 ]; then
+            set_volume 0
+        fi
+        unmute
+    else
+        # Is the volume going down or up?
+        if [ "$delta" -lt 0 ]; then
+            # Down, then we need to make that number positive
+            : $(( delta=delta * -1 ))
+            # And suffix the minus sign.
+            set_volume "${delta}-"
+        else
+            # Only suffix the plus sign.
+            set_volume "${delta}+"
+        fi
+    fi
+}
+
+print_help() {
+    # Prints the usage message.
+    # Can also get a message to show first.
+    [ $# -gt 0 ] && echo "$@"
```

Comment:
print_help should always output to stderr, so you can pull the redirects into the function